include/modem/nrf_modem_lib.rst

```diff
@@ -0,0 +1,135 @@
+The glue layer utilizes this functionality to offload the socket API calls to the Modem library and thus eases the task of porting the networking code to the nRF9160 by providing a wrapper for Modem library's native socket API.
+Examples of such wrapper functions are :c:func:`nrf_socket` and :c:func:`nrf_send`.
+
+The socket offloading functionality in the Modem library glue is implemented in :file:`nrf\\lib\\nrf_modem_lib\\nrf91_sockets.c`.
+
+Modem library socket API sets errnos as defined in :file:`nrf_errno.h`.
+The socket offloading support in the Modem library glue in |NCS| converts those errnos to the errnos that adhere to the selected C library implementation.
+
+The socket offloading functionality is enabled by default.
+To disable the functionality, set the :option:`CONFIG_NET_SOCKETS_OFFLOAD` Kconfig option to ``n`` in your project configuration.
+If you disable the socket offloading functionality, the socket calls will no longer be offloaded to the nRF9160 modem firmware.
+Instead, the calls will be relayed to the native Zephyr TCP/IP implementation.
+This can be useful to switch between an emulator and a real device while running networking code on these devices.
+Note that the even if the socket offloading is disabled, Modem library's own socket APIs such as :c:func:`nrf_socket` and :c:func:`nrf_send` remain available.
+
+OS abstraction layer
+====================
+
+For functioning, the Modem library requires the implementation of an OS abstraction layer, which is an interface over the operating system functionalities such as interrupt setup, threads, and heap.
+The Modem library glue in |NCS| provides an implementation of the Modem library OS abstraction layer using |NCS| components.
+The OS abstraction layer is implemented in the :file:`nrfxlib\\nrf_modem\\include\\nrf_modem_os.c`.
+
+The behavior of the functions in the Modem library glue is dependent on the |NCS| components that are used in their implementation.
+This is relevant for functions such as :c:func:`nrf_modem_os_shm_tx_alloc`, which uses :ref:`Zephyr's Heap implementation <zephyr:heap_v2>` to dynamically allocate memory.
+In this case, the characteristics of the allocations made by these functions depend on the heap implementation by Zephyr.
+
+.. _partition_mgr_integration:
+
+Partition manager integration
+=============================
+
+The Modem library, which runs on the application core, shares an area of RAM memory with the nRF9160 modem core.
+During the initialization, the Modem library accepts the boundaries of this area of RAM and configures the communication with the modem core accordingly.
+
+However, it is the responsibility of the application to reserve that RAM during linking, so that this memory area is not used for other purposes and remain dedicated for use by the Modem library.
+Therefore, the configuration parameters of the application decide the position and the size of the area of RAM dedicated to the Modem library.
```

Comment:
My bad, this sentence was not correct. I think we can remove it, the information that it's trying to convey is explained quite clearly in the sentence below.

Reply:
Removed this.